### PR TITLE
feat(turbine): Warn about uncommitted changes

### PR DESCRIPTION
**Pull request by janelletavares**
## Description of change

Fixes https://github.com/meroxa/turbine-project/issues/70

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [ ]  Unit Tests
- [x]  Tested manually locally
- [ ]  Tested in staging

## Demo
Screenshot from 2022-03-10 11-27-56
Screenshot from 2022-03-10 11-28-10

### CODE REVIEW

**Comment by janelletavares** [comment]
I commented out most of the deploy command to test this

File: cmd/meroxa/root/apps/deploy.go [hunk 1]
```diff
@@ -155,6 +215,8 @@ func (d *Deploy) Execute(ctx context.Context) error {
 		return err
 	}
 
+	d.gitWarnings(ctx)
```
Comment (raulb) [inline cmd/meroxa/root/apps/deploy.go:218]: We should interrupt the deploying process if there are changes untracked. Maybe, if `gitWarnings` returned an error, we could do that.

**Review by raulb — CHANGES_REQUESTED** [review]
There is a couple of required changes so we stop the execution of the deployment once we have untracked changes. 

Aside, I'd like to make the suggestion of simplifying this a bit. What I think could be simpler is:

1. Run `cmd := exec.Command("git", "status", "--porcelain=v2")`
2. If there are untracked changes, then run `git status` and show the output to the user including a custom message indicating that need to commit first.
3. If everything is tracked continue. 

What do you think? @janelletavares

File: cmd/meroxa/root/apps/deploy.go [hunk 2]
```diff
@@ -143,6 +145,64 @@ func (d *Deploy) createApplication(ctx context.Context) error {
...
+		return
+	}
+	err = os.Chdir(d.path)
+	if err != nil {
+		return
+	}
+
+	cmd := exec.Command("git", "status", "--porcelain=v2")
+	output, err := cmd.Output()
+	if err == nil {
```
[the remark this answers is not in the document]
Comment (janelletavares) [reply cmd/meroxa/root/apps/deploy.go:164]: so there's a way to deploy without having called `init`?

File: cmd/meroxa/root/apps/deploy.go (hunk 1 above)
Comment (janelletavares) [reply cmd/meroxa/root/apps/deploy.go:218]: so it's not so much a warning as an error? that does make sense because it is the main branch.

Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:218]: @janelletavares that's correct.

File: cmd/meroxa/root/apps/deploy.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:164]: @janelletavares we should consider that option, yes. Not what we'd expect, but I think that's possible.

**Review by raulb — APPROVED** [review]
👌

**Pull request merged**